### PR TITLE
Domains: Add copy to map or transfer domain in the removal dialog

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR updates the domain removal dialog, to include a copy that lets user know they can map their domain to another site, or they can also transfer it to another provider.

#### Testing instructions

To test this, you need either a domain purchased with credits or a domain outside the refund window.

- Go to Payment Settings for a domain you own
- Click on the Remove domain button and verify by checking against the screenshots below
- Verify that the links in the dialog take you to the correct support page

You should see this dialog (pay attention to both the message and the buttons):

Screenshot 2020-01-10 at 1 43 33 PM

NOTE: if the domain is within refund period, you might see an additional piece of text in the dialog below the changes introduced here. That is expected.

### CODE REVIEW

**Comment by eltongo** [comment]
Great suggestion!

**Comment by daledupreez** [comment]
I checked -- it looks like the test failure is unrelated.

File: client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx [hunk 1]
```diff
@@ -45,17 +63,21 @@ class RemoveDomainDialog extends Component {
 						components: { strong: <strong /> },
 					} ) }
 				</FormSectionHeading>
+
+				{ this.renderDomainDeletionWarning( productName ) }
+
 				<p>
+					{ translate( 'If you want to use this domain with another service, DO NOT delete it.' ) }{ ' ' }
```
Comment (fditrapani) [inline client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: This language is a bit harsh. What if we say:

`If you want to use this domain with another service, you don't have to delete it. Instead, keep the domain and then move or point it to another service.`

Comment (eltongo) [reply client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: A string update would take at least 1 week to translate. I'm going to deploy this, but will bring it up for discussion. Thanks :)

**Pull request merged**